File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
         vector<int>arr(26,0);
         int i=0;
        while(s[i]!='\0'){
               arr[s[i]-'a']++; i++;
        }
        for(int i =0 ; i<s.size();i++){
            if(arr[s[i] - 'a'] == 1){
                return i;
            }
        }
       
         return -1;

    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        LinkedHashMap<Character,Integer>mp = new LinkedHashMap<>(); 
        for(int i=0;i<s.length();i++){
           if(mp.containsKey(s.charAt(i))){
              mp.put(s.charAt(i),mp.getOrDefault(s.charAt(i),0) + 1);
           }
           else mp.put(s.charAt(i),1);
        }

        for(Map.Entry<Character,Integer> m : mp.entrySet()){
            if(m.getValue() == 1) return s.indexOf(m.getKey());
        }
        return -1;

    }
}